iquip/apps/builder.py

```diff
@@ -185,6 +185,49 @@ def value(self) -> str:
         return self.lineEdit.text()
 
 
+class _DateTimeEntry(_BaseEntry):
+    """Entry class for a date and time value.
+    
+    Attributes:
+        checkBox: The checkbox for the availability of the dateTimeEdit.
+        dateTimeEdit: The dateTimeEdit for the date and time value.
+    """
+
+    def __init__(self, name: str, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(name, {}, parent=parent)
+        # widgets
+        self.checkBox = QCheckBox(self)
+        self.checkBox.setChecked(False)
+        self.checkBox.stateChanged.connect(self.updateDateTimeEditState)
+        currentDateTime = QDateTime.currentDateTime()
+        self.dateTimeEdit = QDateTimeEdit(currentDateTime, self)
+        self.dateTimeEdit.setCalendarPopup(True)
+        self.dateTimeEdit.setDisplayFormat("yyyy-MM-dd HH:mm:ss")
+        self.dateTimeEdit.setEnabled(False)
+        self.dateTimeEdit.setMinimumDateTime(currentDateTime)
+        # layout
+        self.layout.addWidget(self.checkBox)
+        self.layout.addWidget(self.dateTimeEdit)
+
+    @pyqtSlot()
+    def updateDateTimeEditState(self):
+        """Enables or disables the dateTimeEdit in according to the state of the checkBox.
+        
+        Once the state of the checkBox is changed, this is called.
+        """
+        self.dateTimeEdit.setEnabled(self.checkBox.isChecked())
+
+    def value(self) -> Optional[str]:
+        """Overridden.
+        
+        Returns the value of the dateTimeEdit in ISO format if it is enabled.
+        Otherwise returns None.
+        """
+        return self.dateTimeEdit.dateTime().toString(Qt.ISODate) if self.dateTimeEdit.isEnabled() \
```

Comment:
Considering the GUI, how about checking `checkBox` instead of `dateTimeEdit`?
Not going to happen but `dateTimeEdit` can be enabled or disabled via codes and the users will see the checkbox to know if the "timed" option is enabled.

Reply:
Oh, it looks more clear! I will apply it.